### PR TITLE
Updated dependencies and fixed bugs

### PR DESCRIPTION
**Pull request by mashaf**
Updated dependencies and fixed bugs since the app wasn't touched for the last 2 years

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -60,7 +60,8 @@
 
         <activity android:name=".splashScreen.SplashScreenActivity"
             android:theme="@style/SplashTheme"
-            android:screenOrientation="portrait">
+            android:screenOrientation="portrait"
+            android:exported="true">
```
Comment (NikolaChachev) [inline app/src/main/AndroidManifest.xml:64]: Are we expecting other apps to be able to launch ours or is there another reason this is set to true ?

Comment (mashaf) [reply app/src/main/AndroidManifest.xml:64]: I think we don't expect that behavior. So, I can safely remove this `android:exported="true"` line?

Comment (NikolaChachev) [reply app/src/main/AndroidManifest.xml:64]: Yup, the exported value should be set to true only if we expect other applications to be able to launch ours. It is perfectly safe to remove it

Comment (mashaf) [inline app/src/main/AndroidManifest.xml:64]: @NikolaChachev When I delete that line, I get this error message when building: 
> Manifest merger failed : android:exported needs to be explicitly specified for element <activity#com.kidssaveocean.fatechanger.splashScreen.SplashScreenActivity>. Apps targeting Android 12 and higher are required to specify an explicit value for `android:exported` when the corresponding component has an intent filter defined. See https://developer.android.com/guide/topics/manifest/activity-element#exported for details.

When I started to research the bug, I found the next point: 
> Starting from Android 12 (API level 31), you must explicitly declare the `android:exported` attribute for activities, services, and broadcast receivers that have intent filters. The system throws an exception if an activity, service, or broadcast receiver that has intent filters does not have this attribute explicitly declared in the app's manifest file.

So, I probably just need to set it `false`?

Comment (mashaf) [reply app/src/main/AndroidManifest.xml:64]: I think, I'll merge this PR and you can fix things later.

**Pull request merged**